### PR TITLE
Remove hardcoded nodeSelector for kubestatematrics

### LINKED ISSUE #3735: helm chart - remove hard coded nodeSelector for kubestatemetrics portion
### Description

If you don't have the node label of control-plane on your cluster the metrics pod will not start.  Consider making this a variable so you can provide your own node selector or provide affinity rules.

https://github.com/volcano-sh/volcano/blob/master/installer/helm/chart/volcano/templates/kubestatemetrics.yaml#L187-L190

### Steps to reproduce the issue

Run the helm chart with  Values.custom.metrics_enable = true

### Describe the results you received and expected

The kube-state-metrics pod should start for clusters that are not using the control-plane label

### What version of Volcano are you using?

volcano version: "1.5" appVersion: "0.1"

### Any other relevant information

_No response_

### PR DESCRIPTION
**Pull request by lekaf974**
Fix #3735

### CODE REVIEW

File: installer/helm/chart/volcano/templates/kubestatemetrics.yaml [hunk 1]
```diff
@@ -1,4 +1,5 @@
 {{- if .Values.custom.metrics_enable }}
+{{ $kube_state_metrics_ns := or .Values.custom.admission_ns .Values.custom.kube_state_metrics_ns }}
```
Comment (Monokaix) [inline installer/helm/chart/volcano/templates/kubestatemetrics.yaml:2]: Why add `Values.custom.admission_ns` here?

Comment (lekaf974) [reply installer/helm/chart/volcano/templates/kubestatemetrics.yaml:2]: looks a bad copy/paste will remove that, also should I set kube_state_metrics_ns default value equals to current value ?

**Comment by Monokaix** [comment]
Please also update `installer/README.md` for new added option.

**Comment by lekaf974** [comment]
> Please also update `installer/README.md` for new added option.

Done

**Pull request merged**